Ensuring multiple servers can be run in the same process

GreenGremlin opened the pull request:
The yargs singleton was causing options from the first server to override options for the second server.

``` javascript
const idpConfigs = [{
    port: 7000,
    ...
}, {
    port: 7001,
    ...
}];

idpConfigs.forEach(idpConfig => {
    runServer(idpConfig);
});
```

Without this change, the second idp will attempt to run on port 7000. This is happening because yargs is a singleton and subsequent calls to `yargs.config` do not override earlier calls.

GreenGremlin commented:
Is there anything I can do to help get this merged?

mcguinness commented:
can you update the readme with examples of using it as a require module with config?  it's starting to get complicated (for me at least) to understand runServer vs main bootstrap.

mcguinness commented:
maybe also include a sample settings file for `--settings` as well.

GreenGremlin commented:
I've updated the README with instructions for using saml-idp as a library. I also cleaned up the existing installation and usage instructions a bit. Let me know if you'd like my to change anything.

Looking at the `--setings` option, it's actually redundant and does that same thing as the `--configFile` option. I'll submit a separate PR to remove it.

GreenGremlin commented:
One side note. Going forward, can you push a git tag for each version you publish to npm? It helps in troubleshooting where things changed, version wise.

mcguinness commented:
Thanks for the update.  I merged, tagged, and published.  I will tag going forward.  This project has grown from my personal test tool to something folks are using more broadly so probably a good idea to put more structure around releases.
